icu4c/source/common/dictbe.cpp

```diff
@@ -1341,6 +1374,21 @@ CjkBreakEngine::divideUpDictionaryRange( UText *inText,
     if ((uint32_t)bestSnlp.elementAti(numCodePts) == kuint32max) {
         t_boundary.addElement(numCodePts, status);
         numBreaks++;
+    } else if (isPhraseBreaking) {
+        t_boundary.addElement(numCodePts, status);
+        numBreaks++;
+        int32_t prevIdx = numCodePts;
+
+        for (int32_t i = prev.elementAti(numCodePts); i > 0; i = prev.elementAti(i)) {
+            int length = prevIdx - i;
+            prevIdx = i;
+            // Skip the breakpoint if it belongs to the particle or Hiragana.
```

Comment:
Since the logic after the if block is very simple, how about flip the if condiction (by adding !  in front) and just put that part inside the new if block instead? 
```
            // Skip the breakpoint if it belongs to the particle or Hiragana.
            if (!fSkipSet.containsKey(inString.tempSubString(i, length))) {
                t_boundary.addElement(i, status);
                numBreaks++;
            }
```

Reply:
I could not understand. Could you specify more clearly?

Reply:
Resolved